src/riscv/lib/src/state_backend/proof_backend.rs

```diff
@@ -41,6 +44,63 @@ pub struct ProofGen<M: ManagerBase> {
     _pd: std::marker::PhantomData<M>,
 }
 
+impl<M: ManagerAlloc> ManagerAlloc for ProofGen<M> {
+    fn allocate_region<E, const LEN: usize>(init_value: [E; LEN]) -> Self::Region<E, LEN> {
+        ProofRegion::bind(M::allocate_region::<E, LEN>(init_value))
+    }
+
+    fn allocate_dyn_region<const LEN: usize>() -> Self::DynRegion<LEN> {
+        ProofDynRegion::bind(M::allocate_dyn_region::<LEN>())
+    }
+}
+
+impl<M: ManagerClone> ManagerClone for ProofGen<M> {
+    fn clone_region<E: 'static + Clone, const LEN: usize>(
+        region: &Self::Region<E, LEN>,
```

Comment:
This only clones the underlying region. What about the rest of the proof region?